resources/metadata/githubPublishRelease.yaml

```
@@ -69,6 +69,9 @@ spec:
           - STEPS
         type: string
         default: "master"
```

I think I'd rather see a pipeline fail, than releasing from a "default" commitish like "master" in whatever case there's no CPE or the CPE doesn't have a value for `git/headCommitId`.

---

Yea that's a problem, because we have lot of users who don't run a GPP and use individual steps alone 
@OliverNocon , @anilkeshav27 what would be your suggestion?

---

I would assume that adding the cpe reference should be sufficient. If it is empty, I would assume the default to take over ...

---

My point was that the default bears a considerable risk to be wrong. Also, I'd consider a breaking pipeline better than a wrong release.

---

I suggest to put the default to  `main`

```suggestion
        default: "main"
```

---

My suggestion would be to eliminate the defaulting. It is likely imho that whoever gets broken pipelines by this should fix it by something other than applying `master` or `main`.

```suggestion
```

---

cc @pre thoughts on the defaulting?

---

If anybody is blindly using the default which references a branch (which may change at any given time), their release is already about to be broken at a random time in future. It is better to fail explicitly and instruct in the error to use a configuration which is safe with concurrent builds. 

My thoughts only - i didnt read the code, but i experienced the concurrency failure using a branch reference.

---

Thanks everyone for the input, Would prefer to continue this discussion is a separate GH issue - https://github.com/SAP/jenkins-library/issues/3962